lib/iris/util.py

```diff
@@ -1471,7 +1471,8 @@ def unify_time_units(cubes):
     Performs an in-place conversion of the time units of all time coords in the
     cubes in a given iterable. One common epoch is defined for each calendar
     found in the cubes to prevent units being defined with inconsistencies
-    between epoch and calendar.
+    between epoch and calendar. During this process, all time coordinates have
+    their data type converted to 64-bit floats.
```

Comment:
Very much preference, might be worth explaining why they need to be the same type?

Reply:
No harm in it - most users will be using this function to prepare their cubes for concatenation anyways. pointing out thats the reason doesnt hurt.